sumfields.php

```diff
@@ -386,18 +386,25 @@ function sumfields_civicrm_triggerInfo(&$info, $tableName) {
       $extra_sql = implode(',', $parts);
       $sql = $pre_sql . $generic_sql . $extra_sql . ' ON DUPLICATE KEY UPDATE ' . $extra_sql . ';' . $post_sql;
 
+      $variables = ["
+        DECLARE calculated_contact_id integer;
+        SET calculated_contact_id := (SELECT contact_id FROM civicrm_contribution WHERE id = NEW.contribution_id);
```

Comment:
@monishdeb aren't SQL variables normally set to start with an `@`

Reply:
@seamuslee001 actually there are two types of variable declaration, the one which starts with @ is a user-defined loosely typed variable (by loosely means can be directly assigned with a value without declaring the type) and can be re-used in other Triggers/Procedure in a session and without @ is a local variable which is a valid in particular trigger and need to mention their type. For more detail please check [here](https://stackoverflow.com/questions/1009954/mysql-variable-vs-variable-whats-the-difference). 

In this case as there are three CREATE/UPDATE/DELETE event trigger, so I thought its better to have a there respective local variable.